src/App/Document.cpp

```diff
@@ -2826,18 +2826,24 @@ int Document::recompute(const std::vector<App::DocumentObject*> &objs, bool forc
     FC_TIME_LOG(t,"Recompute total");
 
     if (!d->_RecomputeLog.empty()) {
-        d->pendingRemove.clear();
         if (!testStatus(Status::IgnoreErrorOnRecompute))
             Base::Console().Error("Recompute failed!\n");
     }
-    else {
-        for(auto &o : d->pendingRemove) {
-            auto obj = o.getObject();
-            if (obj)
-                obj->getDocument()->removeObject(obj->getNameInDocument());
+
+    for (auto doc : GetApplication().getDocuments()) {
+        decltype(doc->d->pendingRemove) objs;
+        objs.swap(doc->d->pendingRemove);
+        for(auto &o : objs) {
+            try {
+                if (auto obj = o.getObject()) {
+                    obj->getDocument()->removeObject(obj->getNameInDocument());
+                }
+            } catch (Base::Exception & e) {
+                e.ReportException()

```

Comment:
Thanks @realthunder , I had to add the semi colon at the end of this line in order to compile:
`                e.ReportException();`

Reply:
Oops